src/connection.ts

```diff
@@ -25,48 +44,50 @@ import WS from '../websocket';
  * - state: Fired when the state of the Wallet changes.
  * - wallet-update: Fired when a new wallet message arrive from the websocket.
  **/
-class Connection extends EventEmitter {
+abstract class Connection extends EventEmitter {
+  // network: 'testnet' or 'mainnet'
+  public network: string;
```

Comment:
Because otherwise the `WalletServiceConnection` and `WalletConnection` classes wouldn't be able to use it as the network would be private to the `Connection` (base) class

Reply:
Should we use `protected` then?

Reply:
Yes we should 😳

Done in 03477495831cfeb02ed5cc0779843bc151586ff9